### PR TITLE
feat: added AccountUnification modal

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* feat: added AccountUnification modal

Screenshot 2023-09-27 at 9 43 46 PM

TODOs:
- [x] Create modals with basic styles
- [x] Add dark theme
- [x] Add mobile styles
- [x] Adjust css naming with BEM

**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [x] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**

- (ex: Add feature A)

**Fixes**

- (ex: Fix validation function)

**Changes**

- (ex: Change document B)

**To-dos**

> \*Feel free to remove this section if it's not applicable

- [ ] (ex: add user list)

### CODE REVIEW

**Comment by impelcrypto** [comment]
@ayumitk Great! 
About the CSS class name, please follow the BEM rule as much as possible.
https://css-tricks.com/bem-101/

File: src/components/header/modals/ModalAccountUnification.vue [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    onUnmounted(() => {
+      window.removeEventListener('resize', onHeightChange);
+    });
+
+    const updateSteps = (step: number) => {
+      currentStep.value = step;
+    };
+
+    const store = useStore();
+    const isH160 = computed(() => store.getters['general/isH160Formatted']);
```
Comment (impelcrypto) [inline src/components/header/modals/ModalAccountUnification.vue:128]:
```suggestion
    const isH160 = computed<boolean>(() => store.getters['general/isH160Formatted']);
```

File: src/components/header/modals/account-unification/UserAccount.vue [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      emit('next');
+    };
+
+    const store = useStore();
+    const { iconWallet } = useWalletIcon();
+    const { currentAccount, currentAccountName } = useAccount();
+    const { t } = useI18n();
+    const { currentNetworkIdx } = useNetworkInfo();
+
+    const isH160 = computed(() => store.getters['general/isH160Formatted']);
```
Comment (impelcrypto) [inline src/components/header/modals/account-unification/UserAccount.vue:105]:
```suggestion
    const isH160 = computed<boolean>(() => store.getters['general/isH160Formatted']);
```

File: src/components/header/modals/account-unification/UserAccount.vue [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+    const copyAddress = () => {
```
Comment (impelcrypto) [inline src/components/header/modals/account-unification/UserAccount.vue:107]:
```suggestion
    const copyAddress = (): void => {
```

File: src/components/header/styles/modal-account-unification.scss [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  @apply tw-rounded-full;
+  &.close {
+    background: transparent;
+    color: $navy-1;
+    border: solid 1px $navy-1;
+  }
+}
+
+.account-name {
+  @apply tw-font-bold tw-flex tw-items-center tw-justify-start tw-text-base;
```
Comment (impelcrypto) [inline src/components/header/styles/modal-account-unification.scss:18]: Sorry but we are not using Tailwind CSS anymore, please use pure CSS or SCSS instead.

**Review by impelcrypto — COMMENTED** [review]
@ayumitk Please take a look at my comments. Thanks!

File: src/components/header/modals/ModalAccountUnification.vue [hunk 5]
```diff
@@ -40,6 +72,25 @@ export default defineComponent({
   setup(props, { emit }) {
     const isSelected = ref<boolean>(false);
     const isClosing = ref<boolean>(false);
+    const currentStep = ref<number>(0);
+
+    const modalTitle = computed(() => {
```
Comment (impelcrypto) [inline src/components/header/modals/ModalAccountUnification.vue:77]: const modalTitle = computed((): string => {

File: src/components/header/modals/ModalAccountUnification.vue [hunk 6]
```diff
@@ -69,12 +120,23 @@ export default defineComponent({
       window.removeEventListener('resize', onHeightChange);
     });
 
+    const updateSteps = (step: number) => {
```
Comment (impelcrypto) [inline src/components/header/modals/ModalAccountUnification.vue:123]: const updateSteps = (step: number): void => {

**Comment by impelcrypto** [comment]
@ayumitk I've pulled the latest feat/account-unification branch

**Comment by ayumitk** [comment]
@impelcrypto I added another modal for selecting NFT ([220a5b2](https://github.com/AstarNetwork/astar-apps/pull/935/commits/220a5b2fd76b8ffc7d9b4d4439a0c0c266674366))

If a user has no NFT, it’ll show the Astar gradient image as a default, like the attached picture.

Screenshot 2023-09-26 at 2 09 21 PM

**Comment by ayumitk** [comment]
@Kahonnohak I added static texts on the modals (attached pic). It seems some texts are placeholder, and some links are not ready yet. Please let me know if I need to comment out for the Shibuya release.

cc: @impelcrypto 

Screenshot 2023-09-27 at 11 28 22 AM

**Comment by Kahonnohak** [comment]
@ayumitk thanks, all text are fine with placeholders and we don't have a documentation ready, so as long as we have a space for it we are good for now

File: src/components/header/modals/account-unification/Help.vue [hunk 7]
```diff
@@ -0,0 +1,33 @@
+<template>
+  <div class="wrapper--help">
+    <div class="text--title">Need help?</div>
+    <dl class="list--help">
+      <dt>[place holder] I don’t know which address I need to use?</dt>
+      <dd>
+        You don’t need to think of which one, balance of the both accounts are the same. You don’t
```
Comment (impelcrypto) [inline src/components/header/modals/account-unification/Help.vue:7]: @ayumitk 
Let's use i18

File: src/components/header/modals/ModalAccountUnification.vue (hunk 1 above)
Comment (ayumitk) [reply src/components/header/modals/ModalAccountUnification.vue]: Fixed it! Thank you for review

File: src/components/header/modals/account-unification/UserAccount.vue (hunk 2 above)
Comment (ayumitk) [reply src/components/header/modals/account-unification/UserAccount.vue]: Fixed it

File: src/components/header/modals/account-unification/UserAccount.vue (hunk 3 above)
Comment (ayumitk) [reply src/components/header/modals/account-unification/UserAccount.vue]: Fixed it

File: src/components/header/styles/modal-account-unification.scss (hunk 4 above)
Comment (ayumitk) [reply src/components/header/styles/modal-account-unification.scss]: Replaced all tailwind styles to pure css

File: src/components/header/modals/ModalAccountUnification.vue (hunk 5 above)
Comment (ayumitk) [reply src/components/header/modals/ModalAccountUnification.vue]: Fixed it

File: src/components/header/modals/ModalAccountUnification.vue (hunk 6 above)
Comment (ayumitk) [reply src/components/header/modals/ModalAccountUnification.vue]: Fixed it

File: src/components/header/styles/modal-account-unification.scss [hunk 8]
```diff
@@ -0,0 +1,422 @@
+@import 'src/css/quasar.variables.scss';
+@import 'src/css/utils.scss';
+
+.wrapper--account-unification {
+  display: flex;
+  flex-direction: column;
+  gap: 3rem;
```
[the remark this answers is not in the document]
Comment (ayumitk) [reply src/components/header/styles/modal-account-unification.scss]: Replaced all rem to px

File: src/components/header/styles/modal-account-unification.scss [hunk 9]
```diff
@@ -0,0 +1,422 @@
...
+  font-weight: 600;
+  height: 52px;
+  border-radius: 9999px;
+  &.close {
+    background: transparent;
+    color: $navy-1;
+    border: solid 1px $navy-1;
+  }
+}
+body.body--dark {
```
[the remark this answers is not in the document]
Comment (ayumitk) [reply src/components/header/styles/modal-account-unification.scss]: Fixed it

**Review by bobo-k2 — APPROVED** [review]
LGTM

File: src/components/header/modals/ModalConnectWallet.vue [hunk 10]
```diff
@@ -186,6 +207,9 @@ export default defineComponent({
     const { currentAccountName, disconnectAccount } = useAccount();
     const isClosing = ref<boolean>(false);
     const { currentNetworkIdx } = useNetworkInfo();
+    const isAccountUnification = computed<boolean>(() => {
+      return !!(currentNetworkIdx.value === endpointKey.SHIBUYA && currentAccountName);
```
Comment (bobo-k2) [inline src/components/header/modals/ModalConnectWallet.vue:211]: A small suggestion. Maybe to enable account unification UI on a local node also

File: src/components/header/styles/modal-account-unification.scss [hunk 11]
```diff
@@ -0,0 +1,487 @@
+@import 'src/css/quasar.variables.scss';
+@import 'src/css/utils.scss';
+
+.wrapper--account-unification {
+  display: flex;
+  flex-direction: column;
+  gap: 48px;
+  @media (max-width: $sm) {
```
Comment (impelcrypto) [inline src/components/header/styles/modal-account-unification.scss:8]: Please fix all the 'max-width' to 'min-width' as we are using it on this repo.

File: src/components/assets/styles/account.scss [hunk 12]
```diff
@@ -100,8 +100,8 @@
   display: flex;
   justify-content: flex-end;
   column-gap: 24px;
-  @media (min-width: $xl) {
-    column-gap: 80px;
+  @media (max-width: $sm) {
```
Comment (impelcrypto) [inline src/components/assets/styles/account.scss:103]: Please fix all the 'max-width' to 'min-width' as we are using it on this repo.

File: src/components/header/modals/ModalConnectWallet.vue (hunk 10 above)
Comment (impelcrypto) [reply src/components/header/modals/ModalConnectWallet.vue]: @bobo-k2 　I've fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/935/commits/522c5c6a120ca98ffdcaeabee7fe633044250c61) commit

**Comment by impelcrypto** [comment]
@ayumitk Please pull the latest branch as I pushed a few commits.

File: src/components/assets/styles/account.scss (hunk 12 above)
Comment (ayumitk) [reply src/components/assets/styles/account.scss:103]: @impelcrypto fixed it ([15bec29](https://github.com/AstarNetwork/astar-apps/pull/935/commits/15bec290973a72bd4d5b9f22347180a99d9c7bbb))

File: src/components/header/styles/modal-account-unification.scss (hunk 11 above)
Comment (ayumitk) [reply src/components/header/styles/modal-account-unification.scss]: @impelcrypto fixed it ([15bec29](https://github.com/AstarNetwork/astar-apps/pull/935/commits/15bec290973a72bd4d5b9f22347180a99d9c7bbb))

File: src/components/header/modals/account-unification/Help.vue (hunk 7 above)
Comment (ayumitk) [reply src/components/header/modals/account-unification/Help.vue]: @impelcrypto I moved all texts to the en-US lang file and use i18n ([4ed7eb1](https://github.com/AstarNetwork/astar-apps/pull/935/commits/4ed7eb1a34ac5511f8b59da16914fb93f7d1e958), [2b6db2a](https://github.com/AstarNetwork/astar-apps/pull/935/commits/2b6db2a39566f3546912a141645f9dcb53f38f83))

**Comment by Kahonnohak** [comment]
@impelcrypto 
- would it be your plan to move unified account functions to the account icon? 
- Also remove AU related buttons from Shiden and Astar please?
- Can we add the tooltip (we used for folding up the utilities) to the account icon? 
Screenshot 2023-09-28 at 21 33 34

**Pull request merged**